components/serialization/json/src/test/java/com/microsoft/kiota/serialization/mocks/TestEntity.java

```diff
@@ -131,7 +131,12 @@ public Map<String, Consumer<ParseNode>> getFieldDeserializers() {
                 put(
                         "myEnum",
                         (n) -> {
-                            setMyEnum(n.getEnumValue(MyEnum.class));
+                            setMyEnum(
+                                    n.getEnumValue(
+                                            value ->
+                                                    (value == null || MyEnum.values().length < 1)
+                                                            ? null
+                                                            : MyEnum.values()[0].forValue(value)));
```

Comment:
I think this is a fair assumption/implementation to get an instance from the enum.

Reply:
why not simply MyEnum::forValue with a static method instead?

Reply:
Because it cannot be tracked by the type system, to use `MyEnum::forValue` we need `forValue` to be a static method.

Reply:
which is used to be until the recent change?

Reply:
Correct, but in this way we can call the `forValue` without going through reflection which is the point of this PR.
The alternative to this lambda would be to generate another static method to return the enum parser, the change will happen in the generator codebase.

Reply:
Sorry, I still don't understand why we can't simply pass the reference to the static method, without having to use reflection? Like we do for error mappings and so

Reply:
Thanks for insisting, I was completely overlooking it, changing things accordingly now